api4/cloud_test.go

```diff
@@ -141,3 +141,91 @@ func Test_getCloudLimits(t *testing.T) {
 		require.Equal(t, *mockLimits.Messages.History, *limits.Messages.History)
 	})
 }
+
+func TestGetCloudUsageForMessages(t *testing.T) {
+	t.Run("no license returns not implemented", func(t *testing.T) {
+		th := Setup(t)
+		defer th.TearDown()
+
+		th.App.Srv().RemoveLicense()
+
+		usage, r, err := th.Client.GetCloudUsageForMessages()
+		require.Error(t, err)
+		require.Nil(t, usage)
+		require.Equal(t, http.StatusNotImplemented, r.StatusCode)
```

Comment:
given [mozilla's notes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/501) on the status, I'm not 100% sure it is the right one, but 1/5 on this, status are usually very subjective.

Reply:
Since getCloudLimits already returns 501 for the same case, that's why I made it consistent.
Otherwise, I think 403 could be more appropriate, as the resource is forbidden for Non-Licensed/Non-Cloud users.
Whichever we choose, we should make it consistent in both APIs.
cc @neallred

Reply:
Ditto :laughing: I did it because most of the other endpoints in this file did it that way. I lean slightly more towards keeping it consistent with the majority of the other handlers but otherwise agree with @vish9812 . If either of you feel strongly enough to change this endpoint and `getCloudLimits` to return 403 and update tests, I wouldn't oppose it, either.

Reply:
as I said, no strong feelings, it usually ends up being a bit subjective, as long as we are consistent it's all good with me :)

Reply:
Please avoid using 5xx codes if possible. 5xx codes are tracked by SRE and security as SLA violations and eventually will end up in your queue to fix :P There is a recent campaign by SRE to clean up any unnecessary 5xx codes to 4xx. I'd suggest to change everything to 403.

Ticket for reference: https://mattermost.atlassian.net/browse/MM-43202

cc @stylianosrigas @DSchalla

Reply:
Noted, sounds good to me.